### DISCUSSION TITLE
How to show (toggle) OverlayPanel from SplitButton?

### DISCUSSION THREAD

**Discussion by liborjelinek** [primefaces/community#475] [PrimeVue]
Hi!
I want to show OverlayPanel from the SplitButton "main" button and items, but can't figure out how :-)

![image](https://github.com/primefaces/community/assets/383059/8f3813b5-3840-4f12-a69d-f80cd901f24d)

![image](https://github.com/primefaces/community/assets/383059/fc8a874e-7f03-4530-8bb5-783a86eeb91b)

### Setup

Template with SplitButton and OverlayPanel.

```html
<SplitButton
  icon="pi pi-plus"
  :model="items"
  @click="newFile"
  id="explorer-new-button" />

<OverlayPanel ref="opNewFile">
  <InputText placeholder="Enter a name"/>
  <Button label="Ok" />
</OverlayPanel>
```

SplitButton's items are

```js
const items: MenuItem[] = [
  { label: "New file...", command: newFile },
  { label: "New folder..", command: newFolder }
]
```

OverlayPanel is reachable as `opNewFile` ref:

```js
const opNewFile = ref()
```

### Show OverlayPanel from SplitButton main button

There is `@click=newFile` on `<SplitButton>` to show OverlayPanel from SplitButton "main button".

```js
function newFile(event) {
  opNewFile.value.toggle(event)
}
```

and it works.

### ... but how to open OverlayPanel from SplitButton items?

Overlay's `toggle()` needs event to get element to attach to. However, inside command function there is no event to pass it to `toggle()`.

By studying sources I discovered that the only required member of event is actually `currentTarget` and tried to cheat OverlayPanel:

```js
function newFile() {
  opNewFile.value.toggle({
    currentTarget: document.querySelector("#explorer-new-button")
  })
}
```

... and it doesn't work :-)

See it yourself at https://codesandbox.io/p/sandbox/frosty-roentgen-8ktp8x

Thanks for any help guys!

**Comment by tugcekucukoglu** [comment]
Click callback is called when the main button is clicked but the menu button does not call any callback. Therefore, opening an OverlayPanel when the menu button is clicked may not be the right approach.

**Reply by liborjelinek** [reply]
Thank you trying to help me 🙏. I understand your point but for my case, I think it makes sense. Anyway, let's leave UX discussion behind.

Showing OverlayPanel should be limited just to button click handler...

I just want to "manually" control showing OverlayPanel - from menu item command function, from some non-event handler function, maybe from watcher, ... Is it somehow possible, please?

**Comment by aasiph** [comment]
This works.
```javascript
function newFile() {
  setTimeout(()=> {
    opNewFile.value.toggle({
      currentTarget: document.querySelector("#explorer-new-button") 
    })
  }, 100)
}
```
